pkg/util/util.go

```diff
@@ -26,3 +34,41 @@ func ContextFromStopChannel(stopCh <-chan struct{}) context.Context {
 
 	return ctx
 }
+
+// DNSNamesForService returns the possible DNS names for a service with the given name and namespace
+func DNSNamesForService(name, namespace string) []string {
```

Comment:
I don't see how all these new methods are general utils for _all_ our controllers - they only make sense for the Control Plane controller (now). I'd rather move them into a `controlplane` package as we did it for `infrastructure` etc.

Reply:
I put them where we agreed with @rfranzke they should be. But I think you have a point, so I moved them to `controlplane`.